src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java

```diff
@@ -0,0 +1,134 @@
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.module.swiftiso20022.constants.ConnectorConstants;
+import org.wso2.carbon.module.swiftiso20022.model.ErrorModel;
+import org.wso2.carbon.module.swiftiso20022.utils.ConnectorUtils;
+import org.wso2.carbon.module.swiftiso20022.utils.MT940ValidationUtils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+
+/**
+ * Class to validate the MT940 format.
+ */
+public class MT940FormatValidator extends AbstractConnector {
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        log.debug("Executing MT940FormatValidator to validate the MT940 format");
+        org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) messageContext)
+                .getAxis2MessageContext();
+        Optional<String> payload = ConnectorUtils.buildMessagePayloadFromMessageContext(axis2MessageContext);
+        if (payload.isPresent()) {
+            String[] lines = payload.get().split("\n");
+            ErrorModel validationResponse = validateMT940(messageContext, lines);
+
+            if (validationResponse.isError()) {
+                this.log.error(validationResponse.getErrorMessage());
+                ConnectorUtils.appendErrorToMessageContext(messageContext, validationResponse.getErrorCode(),
+                        validationResponse.getErrorMessage());
+                this.handleException(ConnectorConstants.ERROR_VALIDATION_FAILED, messageContext);
+            }
+
+        } else {
+            this.log.error(ConnectorConstants.ERROR_MISSING_PAYLOAD);
+            ConnectorUtils.appendErrorToMessageContext(messageContext, ConnectorConstants.MISSING_REQUEST_PAYLOAD,
+                    ConnectorConstants.ERROR_MISSING_PAYLOAD);
+            this.handleException(ConnectorConstants.ERROR_MISSING_PAYLOAD, messageContext);
+        }
+    }
+
+    /**
+     * Method to validate the MT940 format.
+     *
+     * @param messageContext  Message context
+     * @param lines           Payload lines
+     * @return                Error model
+     */
+    private ErrorModel validateMT940(MessageContext messageContext, String[] lines) {
+        ErrorModel errorModel = new ErrorModel();
+        Map<String, Object> extractFields = extractFields(lines);
+
+        if (!MT940ValidationUtils.validateC1Rule(lines)) {
+            return new ErrorModel(ConnectorConstants.ERROR_C24,
+                    ConnectorConstants.ERROR_FIELD_86);
+        }
+
+        if (!MT940ValidationUtils.validateC2Rule(lines)) {
+            return new ErrorModel(ConnectorConstants.ERROR_C277,
+                    ConnectorConstants.ERROR_BALANCES);
+        }
+
+        errorModel = MT940ValidationUtils.validateMT940Format(extractFields);
```

Comment:
When L78 or L83 is true, L75 becomes unnecessary 

```suggestion
        ErrorModel errorModel = MT940ValidationUtils.validateMT940Format(extractFields);
```

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/2/commits/8d6b87e8988955fbe9679e7afa9d4285b0ff0ac8